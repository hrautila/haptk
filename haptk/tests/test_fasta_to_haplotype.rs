mod common;

use crate::common::TEST_FASTA;
use std::path::PathBuf;

#[test]
#[cfg(feature = "clap")]
fn fasta_to_haplotype() {
    let cmd = haptk::clap::SubCommand::FastaToHaplotype {
        file: PathBuf::from(TEST_FASTA),
        seq_name: vec![String::from(
            "chr19:17676011-17676041_ENST00000519716.7_exon_1_0_chr19_17676012_r",
        )],
        log_and_verbosity: crate::common::silent_verbosity(),
        output: PathBuf::from("tests/results/exon_haplotype.csv"),
    };

    haptk::clap::run_cmd(cmd).unwrap();

    let res = std::fs::read_to_string("tests/results/exon_haplotype.csv").unwrap();
    insta::assert_yaml_snapshot!(res);
}
