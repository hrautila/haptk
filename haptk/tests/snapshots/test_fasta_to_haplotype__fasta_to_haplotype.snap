---
source: haptk/tests/test_fasta_to_haplotype.rs
expression: res
---
"chr19,17676011,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676012,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676013,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676014,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676015,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676016,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676017,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676018,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676019,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676020,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676021,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676022,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676023,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676024,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676025,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676026,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676027,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676028,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676029,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676030,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676031,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676032,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676033,C,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676034,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676035,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676036,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676037,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676038,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676039,T,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676040,G,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\nchr19,17676041,A,-,0,ENST00000519716.7_exon_1_0_chr19_17676012_r\n"
